airoha: an7581: add support for kernel 6.12

hurrian opened the pull request:
Enable kernel 6.12 as the testing kernel for airoha.

For kernel 6.12, we are able to do away with many older patches and refresh others to be closer to mainline.

DTS changes are backwards-compatible with kernel 6.6.

Tested to flash and boot on Gemtek W1700K (#17869).

Signed-off-by: Kenneth Kasilag <kenneth@kasilag.me>

andrewjlamarche commented on target/linux/airoha/an7581/config-6.6:
```diff
@@ -231,7 +231,8 @@ CONFIG_NET_FLOW_LIMIT=y
 CONFIG_NET_SELFTESTS=y
 # CONFIG_NET_VENDOR_3COM is not set
-CONFIG_NET_VENDOR_MEDIATEK=y
+CONFIG_NET_VENDOR_AIROHA=y
+# CONFIG_NET_VENDOR_MEDIATEK is not set
```
Is this necessary for 6.12 support?

andrewjlamarche commented on target/linux/generic/backport-6.6/703-v6.10-flow_offload-add-control-flag-checking-helpers.patch:
Is this needed for 6.12 support?

andrewjlamarche commented on target/linux/generic/backport-6.6/704-v6.12-ipv6-Add-ipv6_addr_-cpu_to_be32-be32_to_cpu-helpers.patch:
Is this needed for 6.12 support?

andrewjlamarche commented on target/linux/generic/backport-6.6/705-v6.12-netdevice-add-netdev_tx_reset_subqueue-shorthand.patch:
Is this needed for 6.12 support?

namiltd commented:
To keep a history of patches, use the ` ./scripts/kernel_bump.sh -s 6.6 -t 6.12 -p airoha` command to upgrade from a 6.6 kernel to a 6.12 kernel, then perform further fixes.

namiltd reviewed (commented):
Missing commit message. Please describe your changes

```
airoha: an7581: add support for kernel 6.12
<add here empty line>
Refresh patches and config for kernel 6.12.
```

hurrian commented on target/linux/airoha/an7581/config-6.12:
Thanks! Resolved in [v8/4aea29a](https://github.com/openwrt/openwrt/commit/4aea29ac5f9de54f03af6d370bac6fc7238e2fd5)

abajk commented:
Airoha patches have recently been merged. Rebase this PR on top of the current master.

danpawlik commented:
@hurrian hey, could you rebase the PR and change the PR title please?

hurrian commented:
> @hurrian hey, could you rebase the PR and change the PR title please?
> 
> EDIT: @hurrian I rebased your change on the top of main branch (it is available https://github.com/danpawlik/openwrt/tree/main-w1700k-6.12 + include patch from #17869) but I don't have enough knowledge, what's going on, that Ansuel patch ([200-spinlock-extend-guard-with-spinlock_bh-variants.patch](https://github.com/openwrt/openwrt/commit/a41fda02f26424e4c033c12b671408f0cb8345b4#diff-78d6d1b02b553892a6704afc0002085d68b8596c530c8c5e99a1f0c479e54890) does not work...

I just saw this now, thanks! 

That said I think the spinlock patch can be dropped or replaced with the real mainline version (it got merged sometime 6.15 IIRC?); it's not critical for the device patch set.

danpawlik commented:
something that might be interesting for this PR https://github.com/openwrt/openwrt/pull/19816

hurrian committed:
kernel/airoha: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Kenneth Kasilag <kenneth@kasilag.me>
Link: https://github.com/openwrt/openwrt/pull/19038
Signed-off-by: Robert Marko <robimarko@gmail.com>

hurrian committed:
kernel/airoha: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Kenneth Kasilag <kenneth@kasilag.me>
Link: https://github.com/openwrt/openwrt/pull/19038
Signed-off-by: Robert Marko <robimarko@gmail.com>

hurrian committed:
airoha: an7581: add support for kernel 6.12

Enable kernel 6.12 as the testing kernel for airoha.

This first commit will largely maintain feature-parity with kernel 6.6.

DTS changes are backwards-compatible with kernel 6.6.

Tested to flash and boot on Gemtek W1700K (#17869).

Signed-off-by: Kenneth Kasilag <kenneth@kasilag.me>
Link: https://github.com/openwrt/openwrt/pull/19038
Signed-off-by: Robert Marko <robimarko@gmail.com>

hurrian commented:
> Airoha patches have recently been merged. Rebase this PR on top of the current master.

Thanks! [v8/965d9a3](https://github.com/openwrt/openwrt/commit/965d9a36600cc081f96992b2b19ce951818e1b2c) rebases on top of the current master tree, and hems closer to feature-parity with the current 6.6 rather than the previous patch sets.

andrewjlamarche reviewed (approved):
Tested on Gemtek W1700K, all functionality appears to work as on 6.6. Well done!

```
Tested-by: Andrew LaMarche <andrewjlamarche@gmail.com>
```

danpawlik reviewed (approved):
Works well. Thanks

andrewjlamarche commented:
Ping @hauke @Ansuel @robimarko for visibility, since this was in limbo for a while.

robimarko commented:
Thanks! Rebased on top of main and merged!
